### ISSUE TITLE
Remove variable value for homework questions

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#3618]
As [described in a comment](https://github.com/PrairieLearn/PrairieLearn/issues/2905#issuecomment-751825284), the current Homework scoring system in PrairieLearn uses a slightly "gamified" mechanism whereby the points awarded for a question increase each time the question is correctly answered, and reset each time it is incorrectly answered.

As @jonatanschroeder [points out], this is both undocumented and confusing. I also don't believe that it is adding any significant value.

We should remove this mechanism and always award `assessment_question.points`, rather than `instance_questions.current_value`. We should also remove the `constantQuestionValue` option.

---

As an alternative to the above plan, we could leave all the logic in place and simply change the default to be `constantQuestionValue=true` (it currently defaults to `false`). We could then wait for 6 months (or more) and see whether anyone explicitly switches to `constantQuestionValue=false`.

---

In any case, we will need a news item because this is a change in behavior.

**Comment by jonatanschroeder** [comment]
Since this is change in behaviour anyway, I propose this field is replaced with a field called `increasingQuestionValue` that has the opposite effect.

**Comment by mwest1066** [comment]
> Since this is change in behaviour anyway, I propose this field is replaced with a field called `increasingQuestionValue` that has the opposite effect.

I'm not opposed to that, but we should also keep `constantQuestionValue` for backwards compatibility. I just checked and there are two courses that are using it.

**Comment by mwest1066** [comment]
> I'm not opposed to that, but we should also keep `constantQuestionValue` for backwards compatibility. I just checked and there are two courses that are using it.

To be clear, back-compat options don't need to be documented or referred to in any way. We just want to keep people's existing course configurations working without errors 😄

**Comment by jonatanschroeder** [comment]
> I'm not opposed to that, but we should also keep `constantQuestionValue` for backwards compatibility. I just checked and there are two courses that are using it.

Yeah, that makes sense, and I agree to keep it for back-compat. These two courses are probably not going to be a problem (assuming they are setting it to `true`), the question is how many courses rely on the old default behaviour (`false`) and will need to change it now. I think a separate field where this is explicitly set (referred to in the news item) would probably make this clearer.

**Comment by mwest1066** [comment]
> Yeah, that makes sense, and I agree to keep it for back-compat. These two courses are probably not going to be a problem (assuming they are setting it to `true`), the question is how many courses rely on the old default behaviour (`false`) and will need to change it now. I think a separate field where this is explicitly set (referred to in the news item) would probably make this clearer.

The two existing courses are setting it to `true`, so they are fine.

For other courses, I think we should put in the news item that people only need to be concerned if they have `"type": "Homework"` assessments with the question `maxPoints` at least 5 times the question `points`. If they have this, then they should either:

1. Set the question `maxPoints` to no more than 3 times the question `points` in `"type": "Homework"` assessments. (This is the RECOMMENDED change.)

OR

2. Set `increasingQuestionValue=true` to get the old behavior. (We DO NOT recommend doing this.)

**Comment by jonatanschroeder** [comment]
Margo had a question where maxPoints was 3 times the value of points, and had this issue. So this is a problem in this case as well.

**Comment by mwest1066** [comment]
> Margo had a question where maxPoints was 3 times the value of points, and had this issue. So this is a problem in this case as well.

I'm not quite following what you mean by "is a problem in this case as well"?

**Comment by jonatanschroeder** [comment]
> people only need to be concerned if they have "type": "Homework" assessments with the question maxPoints at least 5 times the question points.

I mean it's times 3, not times 5.

**Comment by mwest1066** [comment]
> I mean it's times 3, not times 5.

Ah, right, got it. And yes, even x3 will see a difference, this is true.

**Comment by jonatanschroeder** [comment]
@mwest1066 given this behaviour is [now documented](https://docs.prairielearn.com/assessment/configuration/#question-points-for-homework-assessments), even [with practical examples](https://docs.prairielearn.com/assessment/configuration/#question-scoring-details-for-homework-assessments), is this still planned? Or can this be closed as not planned?

**Comment by mwest1066** [comment]
I still think we should deprecate this or at least make the default be the non-gamified version.
